hypothesis-python/src/hypothesis/strategies/_internal/core.py

```diff
@@ -43,6 +43,7 @@
 from uuid import UUID
 
 import attr
+from typing_extensions import Concatenate
```

Comment:
I don't think `typing_extensions` is a requirement of Hypothesis, and I'm not sure if you want to make it one. Any ideas how to handle this? Anything I could reuse from `strategies/_internal/types.py`?

Reply:
It's not a requirement, and I'd like to keep it optional - in which case I think the best option is to try importing ParamSpec (from typing, then typing_extensions); and then have two function definitions behind an `if ParamSpec is not None:`/`else` condition.  Probably makes sense to write this as a very simple wrapper for an unannoted `_`-prefixed function so we're at least not duplicating the logic.